yarn-project/validator-client/src/validator.ts

```diff
@@ -232,20 +232,33 @@ export class ValidatorClient extends (EventEmitter as new () => WatcherEmitter)
     const partOfCommittee = inCommittee.length > 0;
 
     const proposalInfo = {
-      slotNumber,
-      blockNumber,
+      ...proposal.toBlockInfo(),
       proposer: proposer.toString(),
-      archive: proposal.payload.archive.toString(),
-      txCount: proposal.payload.txHashes.length,
-      txHashes: proposal.payload.txHashes.map(txHash => txHash.toString()),
     };
-    this.log.info(`Received request to attest for slot ${slotNumber}`, proposalInfo);
+
+    this.log.info(`Received proposal for slot ${slotNumber}`, {
+      ...proposalInfo,
+      txHashes: proposal.payload.txHashes.map(txHash => txHash.toString()),
+    });
+
+    // Collect txs from the proposal. Note that we do this before checking if we have an address in the
+    // current committee, since we want to collect txs anyway to facilitate propagation.
+    const { txs, missingTxs } = await this.txProvider.getTxsForBlockProposal(proposal, {
+      pinnedPeer: proposalSender,
+      deadline: this.getReexecutionDeadline(proposal, this.blockBuilder.getConfig()),
+    });
```

Comment:
Even if the block proposal is invalid? Could this be a griefing vector where I send bogus proposals to my peers filling up their event queue and/or causing them to spam the p2p network destroying their peer score?

Reply:
Oh, I just checked the code and it looks like we validate the block proposal in the p2p_client and in the validator. 

https://github.com/AztecProtocol/aztec-packages/blob/1e384ba71fe2561d04fa7c998bf324be071384db/yarn-project/validator-client/src/validator.ts#L259

and 

https://github.com/AztecProtocol/aztec-packages/blob/1e384ba71fe2561d04fa7c998bf324be071384db/yarn-project/p2p/src/services/libp2p/libp2p_service.ts#L685-L691

run the same code.

Reply:
There are still some validations for the block proposal that I'm unsure where they should live.